urllib3/connectionpool.py

```diff
@@ -648,6 +648,9 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                 e = ProxyError('Cannot connect to proxy.', e)
             elif isinstance(e, (SocketError, HTTPException)):
                 e = ProtocolError('Connection aborted.', e)
+            elif isinstance(e, TimeoutError):
+                if not retries.method_whitelist or method.upper() not in retries.method_whitelist:
+                    raise e
```

Comment:
I think that you should actually aim to put this inside `retries.increment`. Generally speaking that seems to have most of the relevant logic.